src/server/zset_family.cc

```diff
@@ -2396,37 +2396,61 @@ OpResult<MScoreResponse> ZSetFamily::ZGetMembers(CmdArgList args, ConnectionCont
 void ZSetFamily::GeoAdd(CmdArgList args, ConnectionContext* cntx) {
   string_view key = ArgS(args, 0);
 
-  // TODO: to handle options and multiple elements
   ZParams zparams;
+  size_t i = 1;
+  for (; i < args.size() - 1; ++i) {
+    ToUpper(&args[i]);
 
-  string_view longitude = ArgS(args, 1);
-  string_view latitude = ArgS(args, 2);
-  string_view member = ArgS(args, 3);
+    string_view cur_arg = ArgS(args, i);
 
-  // TODO: to remove this check once the TODO above is handled.
-  if (args.size() != 4) {
-    return (*cntx)->SendError(kSyntaxErr);
+    if (cur_arg == "XX") {
+      zparams.flags |= ZADD_IN_XX;  // update only
+    } else if (cur_arg == "NX") {
+      zparams.flags |= ZADD_IN_NX;  // add new only.
+    } else if (cur_arg == "CH") {
+      zparams.ch = true;
+    } else {
+      break;
+    }
+  }
+
+  if ((args.size() - i) % 3 != 0) {
```

Comment:
Here's an idea: can you remove `arg`'s prefix of size `i` (via `args.remove_prefix(i)`)?
I think it would make the code easier.
After you do that, you can check here if `(args.size() % 3) != 0` and below the iteration is simpler as well.
I'd also check `|| !args.empty()` here, as we should also return `kSyntaxError` in that case.

Reply:
I'll add the `|| !args.empty()` condition now 👍 
Regarding the `remove_prefix`: do you think it's better? Was trying to keep my code as similar to the `ZADD` implementation as possible

Reply:
Would you like me to add the `|| !args.empty()` condition to ZADD as well?